helm/fullnode/templates/backup-verify.yaml

```diff
@@ -0,0 +1,91 @@
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  name: {{ include "backup.fullname" . }}-backup-verify
+  labels:
+    {{- include "backup.labels" . | nindent 4 }}
+    app.kubernetes.io/name: backup-verify
+spec:
+  concurrencyPolicy: Replace
+  suspend: {{ not .Values.backup.enable }}
+  schedule: {{ .Values.backup_verify.schedule | quote }}
+  jobTemplate:
+    spec:
+      template:
+        metadata:
+          labels:
+            {{- include "backup.selectorLabels" . | nindent 12 }}
+            app.kubernetes.io/name: backup-verify
+          annotations:
+            seccomp.security.alpha.kubernetes.io/pod: runtime/default
+        spec:
+          restartPolicy: Never
+          terminationGracePeriodSeconds: 0
+          containers:
+          - name: backup-verify
+            # use the same image with the backup sts
+            image: {{ .Values.backup.image.repo }}:{{ .Values.backup.image.tag | default .Values.imageTag }}
+            imagePullPolicy: {{ .Values.backup.image.pullPolicy }}
+            command:
+            - /usr/local/bin/db-backup-verify
+            {{- range $.Values.restore.config.trusted_waypoints }}
+            - --trust-waypoint
+            - {{ . }}
+            {{- end }}
+            - --metadata-cache-dir
+            - /tmp/diem-backup-verify-metadata
+            - command-adapter
+            - --config
+            # use the same config with the backup sts
+            - /opt/diem/etc/{{ .Values.backup.config.location }}.toml
+            env:
+            - name: RUST_LOG
+              value: "debug"
+            - name: RUST_BACKTRACE
+              value: "1"
+            - name: PUSH_METRICS_ENDPOINT
+              value: "{{- include "backup.pushGateway" $ }}/metrics/job/{{- .Values.backup_verify.jobName | default "db_backup_verify" }}"
+            {{- include "backup.backupEnvironment" (dict "config" $.Values.backup.config "era" $.Values.chain.era) | nindent 12 }}
+            {{- with .Values.backup_verify }}
+            resources:
+            {{- toYaml .resources | nindent 14 }}
+            volumeMounts:
+            - name: backup-config
+              mountPath: /opt/diem/etc
+            - name: tmp
+              mountPath: /tmp
+            securityContext:
+              readOnlyRootFilesystem: true
+              allowPrivilegeEscalation: false
+              capabilities:
+                drop:
+                - ALL
+          securityContext:
+            runAsNonRoot: true
+            runAsUser: 6180
```

Comment:
who/what is this?

Reply:
It's just to run the container as non-root. 6180 is also the group and user ID of the `diem` user in safety-rules https://github.com/diem/diem/blob/15a692ef5b19b6b597ad5b45e43e2b594b29f0a8/docker/safety-rules/Dockerfile#L30.

Reply:
That should probably be configurable :)

Reply:
It has to match the uid created in the docker file so it makes sense to hardcode it here.

Reply:
I might be missing something, but I see the `validator` and `safety-rules` dockerfiles create this group, but looks like the image `{{ .Values.backup.image.repo }}:{{ .Values.backup.image.tag | default .Values.imageTag }}` resolves to  `diem/tools:release-1.1_a1abfc34`, whose image does not create that user

https://github.com/diem/diem/blob/15a692ef5b19b6b597ad5b45e43e2b594b29f0a8/docker/tools/Dockerfile

and so `su - diem` would fail for that container, no?

Reply:
That's correct. For backup and restore we just want a non-root uid